src/test/helpers/log_test.cpp

```diff
@@ -1,22 +1,88 @@
 #include "test/helpers/log_test.h"
 
-QList<std::tuple<QtMsgType, QRegularExpression>> logMessagesExpected;
+namespace {
 
-void logCapture(QtMsgType msgType, const QMessageLogContext&, const QString& msg) {
-    for (int i = 0; i < logMessagesExpected.size(); i++) {
-        if (std::get<QtMsgType>(logMessagesExpected[i]) == msgType &&
-                std::get<QRegularExpression>(logMessagesExpected[i])
+QList<std::tuple<QtMsgType, QRegularExpression>> s_logMessagesExpected;
+QtMessageHandler s_oldHandler;
+
+void logCapture(QtMsgType msgType, const QMessageLogContext& context, const QString& msg) {
+    s_oldHandler(msgType, context, msg);
```

Comment:
this statement confuses me. does it initialize `s_oldHandler` or call it? I don't know the type semantics of `QtMessageHandler`. A comment would be nice.

Reply:
Indeed. And I will wrap it into a function.